### ISSUE TITLE
Regex bug in python_info_condaenv_find

### ISSUE THREAD

**Issue by cameronbriar** [rstudio/reticulate#1535]
The regular expression [here](https://github.com/rstudio/reticulate/blob/v1.35.0/R/python-tools.R#L189) added in this [change set](https://github.com/rstudio/reticulate/pull/1154/files) appears incorrect, resulting in failures to resolve the correct Conda environment.

To reproduce (inside a Conda-env enabled workspace):

```
> library(reticulate)
> reticulate::use_condaenv('my-env')
Error in normalizePath(conda, winslash = "/", mustWork = TRUE) : 
  path[1]="# cmd: /opt/conda/bin/mamba create": No such file or directory
```

The code linked above attempts to process `/opt/conda/envs/my-env/conda-meta/history` and, if I'm not mistaken, use a regex `^#\\s+cmd: (.+)\\s+(create|rename)\\s+.*` to pull the script path from a line like `# cmd: /opt/conda/bin/mamba create `. Note the trailing space in this example -- which the regex requires at least one of. **In my environment, there's no trailing space and the regex fails, resulting in the error shown above.**

Without refactoring how this is done, I'd propose the following fix to the regex: `^#\\s+cmd: (.+)\\s+(create|rename).*` (or similar)

I'm aware of two workarounds to this issue:

1. (Preferred) In my environment, I can `sed -i 's/mamba create/mamba create /g' /opt/conda/envs/my-env/conda-meta/history` before running the `reticulate::use_condaenv` method OR
2. Use the method described [here](https://github.com/rstudio/reticulate/issues/1460#issuecomment-1803762448) prior to running the `reticulate` method

```
assignInNamespace("is_conda_python", function(x){ return(FALSE) }, ns="reticulate")
```

I'll seek to submit a pull request when I can -- but wanted to create the issue in case it helps the next person in the meantime.

**Comment by t-kalinowski** [comment]
Can you share the contents of the conda history file and how it was generated?

**Comment by paulimer** [comment]
I have the same issue as @jacp1. When you create a environment not with `conda create -n my-env` but with `conda env create -f my-env.yaml`, `use_condaenv()` fails because of the regex. Instead of returning the path to the conda command it returns `/home/username/miniconda3/bin/conda env`, the regex matching `env` as well.

I am also surprised that this fails, given that creating a environment from a yaml file seems common.

I have written an alternative regex that accounts for creating environments with conda env create:
```r
# note perl = TRUE
# line 189 of python-tools.R
script <- sub("^#\\s+cmd: ([^\\s]+)\\s+(env|create|rename)\\s+.*", "\\1", lines[[1]], perl = TRUE)
```
But as I'm not very comfortable with regexes, and find the whole solution brittle, I hesitate to make a pull request.

**Comment by t-kalinowski** [comment]
Thanks for reporting. 

If you update the regex to this locally, does it work for you?

```r
 script <- sub("^#\\s+cmd: (.+)\\s+(env\\s+)?(create|rename)\\s+.*", "\\1", lines[[1]])
```

**Comment by jacp1** [comment]
Unfortunately I can't check for real as I have wiped the whole server and reinstalled using a different solution. However:

```
lines<-list("# cmd: /localhome/temp2/.conda/bin/conda env create -n data-science-stack-2.11.0 -f /localhome/temp2/data-science-stack/environments/data-science-stack-pinned.yaml")
script <- sub("^#\\s+cmd: (.+)\\s+(env\\s+)?(create|rename)\\s+.*", "\\1", lines[[1]])
script
[1] "/localhome/temp2/.conda/bin/conda env"

```
Seems to argue that this doesn't work. I think that the (.+) after cmd is greedily matching the 'env' part as well. I thought this could be solved by using the non-greedy operator cmd: (.+?) but this doesn't seem to work unless you use perl=TRUE.

**Comment by t-kalinowski** [comment]
Good catch! Looks like we'll want to mark the first capture group non-greedy `(.+?)`:

``` r
lines<- "# cmd: /localhome/temp2/.conda/bin/conda env create -n data-science-stack-2.11.0 -f /localhome/temp2/data-science-stack/environments/data-science-stack-pinned.yaml"
# script <-
sub("^#\\s+cmd: (.+?)\\s+(env\\s+)?(create|rename)\\s+.*", "\\1", lines[[1]], perl = TRUE)
#> [1] "/localhome/temp2/.conda/bin/conda"
```

**Comment by t-kalinowski** [comment]
Fixed with #1543
